src/App.jsx

```diff
@@ -1,34 +1,216 @@
+    const payload = {
+      records: [
+        {
+          id: editedTodo.id,
+          fields: {
+            title: editedTodo.title,
+            isCompleted: editedTodo.isCompleted,
+          },
+        },
+      ],
+    };
+    const options = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: token,
+      },
+      body: JSON.stringify(payload),
+    };
+    try {
+      setIsSaving(true);
+      const resp = await fetch(url, options);
+      if (!resp.ok) {
+        throw new Error(resp.message);
       }
-      return todo;
-    });
-    setTodoList(updatedTodos);
+      const { records } = await resp.json();
+      const updatedTodo = {
+        id: records[0].id,
+        ...records[0].fields,
+      };
+      if (!records[0].fields.isCompleted) {
+        updatedTodo.isCompleted = false;
+      }
+      const updatedTodos = todoList.map((todo) => {
+        if (todo.id === updatedTodo.id) {
+          return { ...updatedTodo };
+        } else {
+          return todo;
+        }
+      });
+      setTodoList([...updatedTodos]);
+    } catch (error) {
+      console.error(error);
+      setErrorMessage(`${error.message}. Reverting todo...`);
+      const revertedTodos = todoList.map((todo) => {
+        if (todo.id === originalTodo.id) {
+          return { ...originalTodo };
+        } else {
+          return todo;
+        }
+      });
+      setTodoList([...revertedTodos]);
+    } finally {
+      setIsSaving(false);
+    }
   }
 
-  function completeTodo(id) {
-    const updatedTodos = todoList.map((todo) => {
-      if (todo.id === id) {
-        let value = true;
-        return { ...todo, isCompleted: value };
+  const completeTodo = async (id) => {
+    console.log('App: completeTodo: async');
+    const originalTodo = todoList.find((todo) => todo.id === id)
+    const payload = {
+      records: [
+        {
+          id: originalTodo.id,
+          fields: {
+            isCompleted: true,
+          },
+        },
+      ],
+    };
+    const options = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: token,
+      },
+      body: JSON.stringify(payload),
+    };
+    try {
+      setIsSaving(true);
+      const resp = await fetch(url, options);
+      if (!resp.ok) {
+        throw new Error(resp.message);
+      }
+      const { records } = await resp.json();
+      const updatedTodo = {
+        id: records[0].id,
+        ...records[0].fields,
+      };
+      if (!records[0].fields.isCompleted) {
+        updatedTodo.isCompleted = false;
       }
-      return todo;
-    });
-    setTodoList(updatedTodos);
+      const updatedTodos = todoList.map((todo) => {
+        if (todo.id === updatedTodo.id) {
+          return { ...updatedTodo };
+        } else {
+          return todo;
+        }
+      });
+      setTodoList([...updatedTodos]);
```

Comment:
You have a similar issue here with removing a completed todo. There is a small lag similar to your update function. Once solution should work for this two to remove the lag.